[OGUI-1701] Implement session-service

Deaponn opened the pull request:
<!--
Thank you for your pull request!

Please fill up one of the checklist below by changing [ ] to [x].
Remove checklist and/or items that do not apply.
-->


#### I have JIRA issue created
- [x] branch and/or PR name(s) includes JIRA ID
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Deaponn committed:
Configure docker compose to wait for previous stages

Configure nginx proxy to communicate with Control backend which is running locally

Configure Prettier (for formatting) and ESLint (for more agressive linting)

Deaponn committed:
Configure docker compose to wait for previous stages

Configure nginx proxy to communicate with Control backend which is running locally

Configure Prettier (for formatting) and ESLint (for more agressive linting)

Deaponn committed:
Implement frontend service for session management using sessionStorage

Deaponn committed:
revert: configuring ESLint and Prettier - commit fbafe80

Deaponn committed:
revert: configuring ESLint and Prettier - commit fbafe80

Deaponn committed:
cherry-pick: apply 9c9337c (chore: fix ESLint for test files)

Deaponn committed:
cherry-pick: apply 9c9337c (chore: fix ESLint for test files)

Deaponn committed:
chore: fix conflicts

Deaponn committed:
chore: change code to comply with eslint rules

Deaponn committed:
revert: change package.json to not be modified in this PR

Deaponn committed:
feature: move proxy from /api to /control

graduta reviewed (changes requested):
While fetching session data in this way, it means that you will have to do this request at the loading of the application and stop any other request from being sent. Thus, delaying the load of the application.

I would recommend you write a fetch wrapper that always check if a token is present in current state (e.g. sessionService.js) and if not, makes the request to server to get it.

If the token is present, then use that one in the query.

You can have a look at the fetchClient.js in the framework which does exactly this for the rest of the GUIs: https://github.com/AliceO2Group/WebUi/blob/dev/Framework/Frontend/js/src/fetchClient.js

graduta reviewed (commented):
Another comment that I have is:
- if the token is to be stored in sessionStorage of the browser, this of course it will help for example when the browser page is refreshed.
- but this will also create an issue, that is you are not tackling the case in which the token is invalid and you need to refresh the token.

Instead, if you save the token the state of the app, this will be refreshed automatically at each page entire load

This comment is open for discussion

Deaponn committed:
feat: implement testing using TypeScript

Deaponn committed:
feat: implement testing using TypeScript

Deaponn committed:
style: fix formatting

Deaponn committed:
Merge branch 'dev' into feature/CNF/OGUI-1701/session-service

Deaponn committed:
Merge branch 'dev' into feature/CNF/OGUI-1701/session-service

Deaponn committed:
Merge branch 'dev' into feature/CNF/OGUI-1701/session-service

Deaponn committed:
build: fix package-lock.json

Deaponn committed:
build: fix package.json and package-lock.json

Deaponn committed:
build: fix package.json and package-lock.json

Deaponn committed:
docs: add JSDoc comments

Deaponn committed:
test: update tests and move them to correct directory

Deaponn committed:
test: update tests and move them to correct directory

Deaponn committed:
style: fix formatting

graduta committed:
[release/cog/2.0.0] (#3033)

graduta committed:
[release/cog/2.0.1] (#3035)

dependabot[bot] committed:
Bump puppeteer in /QualityControl in the dev-dependencies group (#3032)

Bumps the dev-dependencies group in /QualityControl with 1 update: [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `puppeteer` from 24.17.0 to 24.18.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.17.0...puppeteer-v24.18.0)

---
updated-dependencies:
- dependency-name: puppeteer
  dependency-version: 24.18.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump puppeteer in /InfoLogger in the dev-dependencies group (#3031)

Bumps the dev-dependencies group in /InfoLogger with 1 update: [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `puppeteer` from 24.17.0 to 24.18.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.17.0...puppeteer-v24.18.0)

---
updated-dependencies:
- dependency-name: puppeteer
  dependency-version: 24.18.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump @stylistic/eslint-plugin from 4.4.1 to 5.3.1 in /InfoLogger (#3030)

Bumps [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic/tree/HEAD/packages/eslint-plugin) from 4.4.1 to 5.3.1.
- [Release notes](https://github.com/eslint-stylistic/eslint-stylistic/releases)
- [Changelog](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint-stylistic/eslint-stylistic/commits/v5.3.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@stylistic/eslint-plugin"
  dependency-version: 5.3.1
  dependency-type: direct:development
  update-type: version-update:semver-major
...

dependabot[bot] committed:
Bump puppeteer in /Control in the dev-dependencies group (#3029)

Bumps the dev-dependencies group in /Control with 1 update: [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `puppeteer` from 24.17.0 to 24.18.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.17.0...puppeteer-v24.18.0)

---
updated-dependencies:
- dependency-name: puppeteer
  dependency-version: 24.18.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump @stylistic/eslint-plugin from 4.4.1 to 5.3.1 in /Framework (#3028)

Bumps [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic/tree/HEAD/packages/eslint-plugin) from 4.4.1 to 5.3.1.
- [Release notes](https://github.com/eslint-stylistic/eslint-stylistic/releases)
- [Changelog](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint-stylistic/eslint-stylistic/commits/v5.3.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@stylistic/eslint-plugin"
  dependency-version: 5.3.1
  dependency-type: direct:development
  update-type: version-update:semver-major
...

dependabot[bot] committed:
Bump actions/checkout from 4 to 5 (#3025)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

dependabot[bot] committed:
Bump google-protobuf from 3.21.0 to 4.0.0 in /Control (#3006)

Bumps [google-protobuf](https://github.com/protocolbuffers/protobuf-javascript) from 3.21.0 to 4.0.0.
- [Release notes](https://github.com/protocolbuffers/protobuf-javascript/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf-javascript/blob/main/protobuf_javascript_release.bzl)
- [Commits](https://github.com/protocolbuffers/protobuf-javascript/compare/v3.21.0...v4.0.0)

---
updated-dependencies:
- dependency-name: google-protobuf
  dependency-version: 4.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

dependabot[bot] committed:
Bump the dev-dependencies group across 1 directory with 2 updates (#3037)

Bumps the dev-dependencies group with 2 updates in the /Framework directory: [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `eslint-plugin-jsdoc` from 54.1.0 to 54.3.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v54.1.0...v54.3.1)

Updates `puppeteer` from 24.17.0 to 24.18.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.17.0...puppeteer-v24.18.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 54.3.1
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.18.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

mariscalromeroalejandro committed:
[OGUI-1722] Runs mode backend: adapt for frontend support  (#3022)

This PR updates the backend to support the frontend in utilizing the `Runs Mode` functionality. Changes include:

- Added a new endpoint to fetch the status of runs by providing the run number.
- Implemented middlewares for validation in controllers when a run number is provided.
- Updated tests to cover the new endpoint and validation logic.

mariscalromeroalejandro committed:
[OGUI-1722] Integrate runs mode functionality (#3023)

This PR updates the frontend pages (objects tree list view, object view and layout view) to integrate with the new backend support for Runs Mode. Changes include:

- Added logic to query the run status using the new backend endpoint.
- Implemented input validation for run numbers before sending requests.
- Updated UI components to display run status and handle loading/error states.
- Tests

graduta committed:
[release/qcg/3.14.0] (#3042)

* Bump QCG version
* Fix one vulnerability on dependency

Houwie7000 committed:
[OGUI-1757] Update layouts controller to allow objects path (#3038)

* `getLayoutsHandler` now accepts object_path as query parameter
* Introduced new Filter object which can contain the objectPath for seaching layouts.

Houwie7000 committed:
[OGUI-1758] object path filtering in repository layer (#3041)

* Repository of layout method, `listLayouts` now accepts objectPath parameter. If defined this parameter will be used to filter layouts by object_paths

dependabot[bot] committed:
Bump puppeteer in /Control in the dev-dependencies group (#3044)

Bumps the dev-dependencies group in /Control with 1 update: [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `puppeteer` from 24.18.0 to 24.19.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.18.0...puppeteer-v24.19.0)

---
updated-dependencies:
- dependency-name: puppeteer
  dependency-version: 24.19.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

mariscalromeroalejandro committed:
[OGUI-1762] Improve user messages when accessing runs mode (#3049)

* New ERROR status to display error notifications when the run number set is not allowed
* Warning notifications for: not found, bookkeeping unavailable or unknown statuses
* Remove the double validation made to run numbers and apply only validation using the DTO in one middleware
* Runs mode middleware will only check if the run number is provided when in run mode as it will be validated previously by the ObjectsGetDTO if provided

graduta committed:
[OGUI-1748] QC should be using logging that has facility set per process env (#3052)

* [OGUI-1748] Logging to use process.env in QCG and documentation (#3008)

dependabot[bot] committed:
Bump the dev-dependencies group in /InfoLogger with 4 updates (#3047)

Bumps the dev-dependencies group in /InfoLogger with 4 updates: [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js), [eslint](https://github.com/eslint/eslint), [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `@eslint/js` from 9.34.0 to 9.35.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/commits/v9.35.0/packages/js)

Updates `eslint` from 9.34.0 to 9.35.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.34.0...v9.35.0)

Updates `eslint-plugin-jsdoc` from 54.1.0 to 55.0.3
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v54.1.0...v55.0.3)

Updates `puppeteer` from 24.18.0 to 24.19.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.18.0...puppeteer-v24.19.0)

---
updated-dependencies:
- dependency-name: "@eslint/js"
  dependency-version: 9.35.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint
  dependency-version: 9.35.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 55.0.3
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.19.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: George Raduta <george.raduta@cern.ch>

dependabot[bot] committed:
Bump the dev-dependencies group in /QualityControl with 4 updates (#3045)

Bumps the dev-dependencies group in /QualityControl with 4 updates: [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js), [eslint](https://github.com/eslint/eslint), [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `@eslint/js` from 9.34.0 to 9.35.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/commits/v9.35.0/packages/js)

Updates `eslint` from 9.34.0 to 9.35.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.34.0...v9.35.0)

Updates `eslint-plugin-jsdoc` from 54.1.0 to 55.0.3
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v54.1.0...v55.0.3)

Updates `puppeteer` from 24.18.0 to 24.19.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.18.0...puppeteer-v24.19.0)

---
updated-dependencies:
- dependency-name: "@eslint/js"
  dependency-version: 9.35.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint
  dependency-version: 9.35.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 55.0.3
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.19.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump the dev-dependencies group in /Framework with 4 updates (#3046)

Bumps the dev-dependencies group in /Framework with 4 updates: [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js), [eslint](https://github.com/eslint/eslint), [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `@eslint/js` from 9.34.0 to 9.35.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/commits/v9.35.0/packages/js)

Updates `eslint` from 9.34.0 to 9.35.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.34.0...v9.35.0)

Updates `eslint-plugin-jsdoc` from 54.3.1 to 55.0.3
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v54.3.1...v55.0.3)

Updates `puppeteer` from 24.18.0 to 24.19.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.18.0...puppeteer-v24.19.0)

---
updated-dependencies:
- dependency-name: "@eslint/js"
  dependency-version: 9.35.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint
  dependency-version: 9.35.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 55.0.3
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.19.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump vite from 6.3.5 to 6.3.6 in /Configuration/webapp (#3051)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.3.5 to 6.3.6.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.3.6/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.3.6/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 6.3.6
  dependency-type: direct:development
...

graduta committed:
[qcg] Refactor/remove unused refresh timer (#3053)

* removes dead code from the old Online Mode of QCG, this has been replaced by the RunMode

dependabot[bot] committed:
Bump puppeteer in /Control in the dev-dependencies group (#3055)

Bumps the dev-dependencies group in /Control with 1 update: [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `puppeteer` from 24.19.0 to 24.21.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.19.0...puppeteer-v24.21.0)

---
updated-dependencies:
- dependency-name: puppeteer
  dependency-version: 24.21.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump the dev-dependencies group in /QualityControl with 3 updates (#3057)

Bumps the dev-dependencies group in /QualityControl with 3 updates: [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc), [globals](https://github.com/sindresorhus/globals) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `eslint-plugin-jsdoc` from 55.0.3 to 57.2.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v55.0.3...v57.2.1)

Updates `globals` from 16.3.0 to 16.4.0
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v16.3.0...v16.4.0)

Updates `puppeteer` from 24.19.0 to 24.21.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.19.0...puppeteer-v24.21.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 57.2.1
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: globals
  dependency-version: 16.4.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.21.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump the dev-dependencies group in /InfoLogger with 3 updates (#3056)

Bumps the dev-dependencies group in /InfoLogger with 3 updates: [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc), [globals](https://github.com/sindresorhus/globals) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `eslint-plugin-jsdoc` from 55.0.3 to 57.2.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v55.0.3...v57.2.1)

Updates `globals` from 16.3.0 to 16.4.0
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v16.3.0...v16.4.0)

Updates `puppeteer` from 24.19.0 to 24.21.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.19.0...puppeteer-v24.21.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 57.2.1
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: globals
  dependency-version: 16.4.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.21.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump the dev-dependencies group in /Framework with 3 updates (#3058)

Bumps the dev-dependencies group in /Framework with 3 updates: [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc), [globals](https://github.com/sindresorhus/globals) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `eslint-plugin-jsdoc` from 55.0.3 to 57.2.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v55.0.3...v57.2.1)

Updates `globals` from 16.3.0 to 16.4.0
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v16.3.0...v16.4.0)

Updates `puppeteer` from 24.19.0 to 24.21.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.19.0...puppeteer-v24.21.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 57.2.1
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: globals
  dependency-version: 16.4.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.21.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

graduta commented on Configuration/webapp/app/routes/runs/details.tsx:
```diff
@@ -21,15 +21,15 @@ import { RunLogs } from '~/components/runs/run-logs';
 
 const getRun = async (runNumber: number): Promise<Run> => {
-  const response = await fetch(`http://localhost:8080/api/runs/${runNumber}`);
+  const response = await fetch(`http://localhost:8080/control/api/runs/${runNumber}`);
```
It is best to keep only the path and parameters to the request instead of the host and port full path. 
For example, this will work the same
```js
const response = await fetch(`control/api/runs/${runNumber}`);
```

graduta commented on Configuration/webapp/app/routes/runs/details.tsx:
```diff
@@ -21,15 +21,15 @@ import { RunLogs } from '~/components/runs/run-logs';
 
 const getLogs = async (runNumber: number): Promise<Log[]> => {
-  const response = await fetch(`http://localhost:8080/api/runs/${runNumber}/logs`);
+  const response = await fetch(`http://localhost:8080/control/api/runs/${runNumber}/logs`);
```
Same as above

graduta commented on Configuration/webapp/app/services/session.ts:
```diff
@@ -0,0 +1,47 @@
+  // only to get the data from server redirect
+  // (line 264, commit 3ba4600 of github.com/AliceO2Group/WebUi/blob/dev/Framework/Backend/http/server.js)
+  // this should be replaced with endpoint designed for authentication only
+  const response = await fetch('http://localhost:8080/control');
```
You should not be using the `host` and `port` in the request as these can change depending on environment. The browser native method `fetch` is smart enough to figure those by itself.
It will work the same with:
```js
const response = await fetch('control');
```

Deaponn replied:
This makes the following error occur while running tests:
```ts
1) `pageRoot` test-suite
       should load session data into memory:
     TypeError: Failed to parse URL from control
      at node:internal/deps/undici/undici:13510:13
      at async fetchSessionData (app/services/session.ts:24:20)
      at async getSessionData (app/services/session.ts:37:5)
```

graduta commented on Configuration/webapp/app/root.tsx:
```diff
@@ -36,6 +38,10 @@ import MainLayout from './components/layout/MainLayout';
  * @returns {React.ReactElement} Root
  */
 export function Layout({ children }: { children: React.ReactNode }) {
+  useEffect(() => {
```
What do you think about not allowing the app to load until the token is fetched? In the end if there is no token, no HTTP endpoint will provide the data needed to build the UI

Deaponn replied:
I was considering implementing the session service as a custom hook which would work like so:
```ts
const { sessionData, isLoading, error } = useSessionData();
```
and then the UI can be built differently based on if the data is still loading or not, or if error is present or not. However I had troubles implementing tests for something like this, but I will work on it

graduta reviewed (changes requested):
The session token is loaded as expected with some minor considerations to be made.
Will you also add the wrapper for the `fetch` requests that appends the token automatically in this PR or different one?

graduta committed:
[OGUI-1769] Keep in-memory information from user's request (#3062)

* adds static information from the user's request to the environment cache object so that it can be displayed to the user during the virtual state PENDING.

Houwie7000 committed:
[OGUI-1759] /layoutes endpoint now accepts filter query param (#3050)

* listLayouts now accepts filter parameter as generalized query param

---------

Co-authored-by: George Raduta <george.raduta@cern.ch>

mariscalromeroalejandro committed:
[OGUI-1766] Run mode checkout (#3054)

This PR updates the runs mode functionality by replacing the previous activation method with a checkbox, making it more intuitive for users to enter runs mode.

mariscalromeroalejandro committed:
[OGUI-1767] Runs mode object tree smart refresh + header refactor (#3066)

This PR provides with a smart refresh in which the objects tree list is refreshed only if new paths need to be listed. It also includes a refactor of the headers for a better styling.

dependabot[bot] committed:
Bump @stylistic/eslint-plugin from 5.3.1 to 5.4.0 in /Framework (#3074)

Bumps [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic/tree/HEAD/packages/eslint-plugin) from 5.3.1 to 5.4.0.
- [Release notes](https://github.com/eslint-stylistic/eslint-stylistic/releases)
- [Changelog](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint-stylistic/eslint-stylistic/commits/v5.4.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@stylistic/eslint-plugin"
  dependency-version: 5.4.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @stylistic/eslint-plugin from 5.3.1 to 5.4.0 in /InfoLogger (#3075)

Bumps [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic/tree/HEAD/packages/eslint-plugin) from 5.3.1 to 5.4.0.
- [Release notes](https://github.com/eslint-stylistic/eslint-stylistic/releases)
- [Changelog](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint-stylistic/eslint-stylistic/commits/v5.4.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@stylistic/eslint-plugin"
  dependency-version: 5.4.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump the dev-dependencies group in /QualityControl with 4 updates (#3072)

Bumps the dev-dependencies group in /QualityControl with 4 updates: [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js), [eslint](https://github.com/eslint/eslint), [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `@eslint/js` from 9.35.0 to 9.36.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/commits/v9.36.0/packages/js)

Updates `eslint` from 9.35.0 to 9.36.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.35.0...v9.36.0)

Updates `eslint-plugin-jsdoc` from 57.2.1 to 60.1.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v57.2.1...v60.1.1)

Updates `puppeteer` from 24.21.0 to 24.22.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.21.0...puppeteer-v24.22.0)

---
updated-dependencies:
- dependency-name: "@eslint/js"
  dependency-version: 9.36.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint
  dependency-version: 9.36.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 60.1.1
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.22.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @grpc/grpc-js from 1.13.0 to 1.14.0 in /Control in the grpc group (#3070)

Bumps the grpc group in /Control with 1 update: [@grpc/grpc-js](https://github.com/grpc/grpc-node).


Updates `@grpc/grpc-js` from 1.13.0 to 1.14.0
- [Release notes](https://github.com/grpc/grpc-node/releases)
- [Commits](https://github.com/grpc/grpc-node/compare/@grpc/grpc-js@1.13.0...@grpc/grpc-js@1.14.0)

---
updated-dependencies:
- dependency-name: "@grpc/grpc-js"
  dependency-version: 1.14.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: grpc
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: George Raduta <george.raduta@cern.ch>

dependabot[bot] committed:
Bump puppeteer in /Control in the dev-dependencies group (#3069)

Bumps the dev-dependencies group in /Control with 1 update: [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `puppeteer` from 24.21.0 to 24.22.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.21.0...puppeteer-v24.22.0)

---
updated-dependencies:
- dependency-name: puppeteer
  dependency-version: 24.22.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: George Raduta <george.raduta@cern.ch>

dependabot[bot] committed:
Bump the dev-dependencies group in /Framework with 4 updates (#3071)

Bumps the dev-dependencies group in /Framework with 4 updates: [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js), [eslint](https://github.com/eslint/eslint), [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `@eslint/js` from 9.35.0 to 9.36.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/commits/v9.36.0/packages/js)

Updates `eslint` from 9.35.0 to 9.36.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.35.0...v9.36.0)

Updates `eslint-plugin-jsdoc` from 57.2.1 to 60.1.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v57.2.1...v60.1.1)

Updates `puppeteer` from 24.21.0 to 24.22.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.21.0...puppeteer-v24.22.0)

---
updated-dependencies:
- dependency-name: "@eslint/js"
  dependency-version: 9.36.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint
  dependency-version: 9.36.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 60.1.1
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.22.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: George Raduta <george.raduta@cern.ch>

dependabot[bot] committed:
Bump the dev-dependencies group in /InfoLogger with 4 updates (#3073)

Bumps the dev-dependencies group in /InfoLogger with 4 updates: [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js), [eslint](https://github.com/eslint/eslint), [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `@eslint/js` from 9.35.0 to 9.36.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/commits/v9.36.0/packages/js)

Updates `eslint` from 9.35.0 to 9.36.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.35.0...v9.36.0)

Updates `eslint-plugin-jsdoc` from 57.2.1 to 60.1.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v57.2.1...v60.1.1)

Updates `puppeteer` from 24.21.0 to 24.22.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.21.0...puppeteer-v24.22.0)

---
updated-dependencies:
- dependency-name: "@eslint/js"
  dependency-version: 9.36.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint
  dependency-version: 9.36.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 60.1.1
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.22.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: George Raduta <george.raduta@cern.ch>

Houwie7000 committed:
[OGUI-1760] Update search UI for object path and add filters (#3065)

* Adds Search/filter layout functionality in Model for layoutList Page
* Adds a SearchModel to be used with registered filter parameters

graduta committed:
[OGUI-329] Add Kafka from ECS Syncronizer in QCG (#3068)

* Adds AliEcsSyncronizer in QCG to consume Kafka Events from ECS as proto objects
* Add internal EventEmitter to update ongoing runs stored by QCG
* Adds `kafkajs` as dependency

mariscalromeroalejandro committed:
[OGUI-1768] Run mode: smart refresh for object view(s) (#3067)

* This PR adds a new check to the object view page and the object tree visualization. When in run mode with an active run, the view will now refresh only when necessary (specifically, if the ID of the last fetch differs from the currently selected ID)

mariscalromeroalejandro committed:
[OGUI-1722] Runs mode layout smart refresh (#3078)

This PR removes the unnecessary objects refresh in layout view by checking their ids and comparing them with the previous one loaded (in run mode) when a run is ongoing. It also removes the property of run status of being RemoteData, so that we avoid an unnecessary blink when it changes.

dependabot[bot] committed:
Bump vite from 6.3.5 to 6.3.6 in /Tokenization/webapp (#3060)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.3.5 to 6.3.6.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.3.6/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.3.6/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 6.3.6
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

mariscalromeroalejandro committed:
[OGUI-329] Adapt frontend to kafka events (#3079)

This PR adapts the frontend to use a dropdown with the ongoing runs provided by Kafka in runs mode.

graduta committed:
Patch release for COG to fix an issue with undefined users (#3081)

graduta committed:
[release/qcg/3.15.0] (#3082)

* Bump QCG

mariscalromeroalejandro committed:
[OGUI-1774] Bugfix: get object by ID properly if in object view from a layout (#3084)

* get object by name or id, based on a context
* patch qcg to v3.15.1

mariscalromeroalejandro committed:
[OGUI-1782] Bug in safari: error using filter property by an iterator (#3095)

* bugfix: convert iterator into an array to be supported by safari
* Bump QCG from 3.15.1 to 3.15.2

dependabot[bot] committed:
Bump nock in /Control in the dev-dependencies group (#3086)

Bumps the dev-dependencies group in /Control with 1 update: [nock](https://github.com/nock/nock).


Updates `nock` from 14.0.0 to 15.0.0
- [Release notes](https://github.com/nock/nock/releases)
- [Changelog](https://github.com/nock/nock/blob/main/CHANGELOG.md)
- [Commits](https://github.com/nock/nock/compare/v14.0.0...v15.0.0)

---
updated-dependencies:
- dependency-name: nock
  dependency-version: 15.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump winston from 3.17.0 to 3.18.0 in /Framework (#3090)

Bumps [winston](https://github.com/winstonjs/winston) from 3.17.0 to 3.18.0.
- [Release notes](https://github.com/winstonjs/winston/releases)
- [Changelog](https://github.com/winstonjs/winston/blob/master/CHANGELOG.md)
- [Commits](https://github.com/winstonjs/winston/compare/v3.17.0...v3.18.0)

---
updated-dependencies:
- dependency-name: winston
  dependency-version: 3.18.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: George Raduta <george.raduta@cern.ch>

dependabot[bot] committed:
Bump eslint-plugin-jsdoc in /InfoLogger in the dev-dependencies group (#3088)

Bumps the dev-dependencies group in /InfoLogger with 1 update: [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc).


Updates `eslint-plugin-jsdoc` from 60.2.0 to 60.5.0
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v60.2.0...v60.5.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 60.5.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump the dev-dependencies group across 1 directory with 3 updates (#3097)

Bumps the dev-dependencies group with 3 updates in the /Framework directory: [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc), [nock](https://github.com/nock/nock) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `eslint-plugin-jsdoc` from 60.2.0 to 60.7.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v60.2.0...v60.7.1)

Updates `nock` from 14.0.0 to 15.0.0
- [Release notes](https://github.com/nock/nock/releases)
- [Changelog](https://github.com/nock/nock/blob/main/CHANGELOG.md)
- [Commits](https://github.com/nock/nock/compare/v14.0.0...v15.0.0)

Updates `puppeteer` from 24.22.1 to 24.23.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.22.1...puppeteer-v24.23.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 60.7.1
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: nock
  dependency-version: 15.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.23.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

dependabot[bot] committed:
Bump the dev-dependencies group across 1 directory with 3 updates (#3096)

* Bump the dev-dependencies group across 1 directory with 3 updates

Bumps the dev-dependencies group with 3 updates in the /QualityControl directory: [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc), [nock](https://github.com/nock/nock) and [puppeteer](https://github.com/puppeteer/puppeteer).


Updates `eslint-plugin-jsdoc` from 60.1.1 to 60.7.1
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Changelog](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.releaserc)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v60.1.1...v60.7.1)

Updates `nock` from 14.0.4 to 15.0.0
- [Release notes](https://github.com/nock/nock/releases)
- [Changelog](https://github.com/nock/nock/blob/main/CHANGELOG.md)
- [Commits](https://github.com/nock/nock/compare/v14.0.4...v15.0.0)

Updates `puppeteer` from 24.22.0 to 24.23.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v24.22.0...puppeteer-v24.23.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-version: 60.7.1
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
- dependency-name: nock
  dependency-version: 15.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: puppeteer
  dependency-version: 24.23.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update docker deps

* Update jsroot patch

---------

Co-authored-by: George Raduta <george.raduta@cern.ch>

dependabot[bot] committed:
Bump actions/setup-node from 4 to 5 (#3092)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 4 to 5.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

graduta committed:
[OGUI-1761] Restrict Lock force ALL actions to admin users only (#3102)

* updates the front-end to display lock buttons for 'Force Release ALL'&'Force Take ALL' to admin users only
* updates the API endpoint to restrict those actions via the middleware
* add tests for aforementioned cases

Reason:
* restriction is needed to prevent shifters from force taking all locks and attempting to start a run with detectors that should not be used. 
* in this manner the shifter has to take the locks one by one, confirming which ones are to be used with SL

Deaponn committed:
refactor: session service

Deaponn committed:
Merge branch 'dev' into feature/CNF/OGUI-1701/session-service

Deaponn committed:
fix: invocation of undefined function

Deaponn committed:
Merge branch 'feature/CNF/OGUI-1701/session-service' of github.com:AliceO2Group/WebUi into feature/CNF/OGUI-1701/session-service

Deaponn committed:
Merge branch 'dev' into feature/CNF/OGUI-1701/session-service

Deaponn commented:
The wrapper for the fetch utility (and the refactor from `sessionService` to `useSession` custom hook) will be developed in another PR. I have already created Jira task for that

Deaponn committed:
Merge branch 'dev' into feature/CNF/OGUI-1701/session-service
